secure_tunneling/include/aws/iotsecuretunneling/SecureTunnel.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/crt/io/Bootstrap.h>
+#include <aws/crt/io/SocketOptions.h>
+#include <aws/iotdevice/secure_tunneling.h>
+#include <aws/iotsecuretunneling/Exports.h>
+
+int s_SecureTunnelingCallbackTest(Aws::Crt::Allocator *allocator, void *ctx);
+
+namespace Aws
+{
+    namespace Iotsecuretunneling
+    {
+        class AWS_IOTSECURETUNNELING_API DeviceApiHandle final
+        {
+          public:
+            DeviceApiHandle(Crt::Allocator *allocator) noexcept;
+            ~DeviceApiHandle();
+            DeviceApiHandle(const DeviceApiHandle &) = delete;
+            DeviceApiHandle(DeviceApiHandle &&) = delete;
+            DeviceApiHandle &operator=(const DeviceApiHandle &) = delete;
+            DeviceApiHandle &operator=(DeviceApiHandle &&) = delete;
+        };
+
+        // Client callback type definitions
+        using OnSecureTunnelingConnectionComplete = std::function<void(void)>;
+        using OnSecureTunnelingSendDataComplete = std::function<void(int errorCode)>;
+        using OnSecureTunnelingDataReceive = std::function<void(const Crt::ByteBuf &data)>;
+        using OnSecureTunnelingStreamStart = std::function<void()>;
+        using OnSecureTunnelingStreamReset = std::function<void(void)>;
+        using OnSecureTunnelingSessionReset = std::function<void(void)>;
+
+        class AWS_IOTSECURETUNNELING_API SecureTunnel final
+        {
+          public:
+            SecureTunnel(
+                // Parameters map to aws_secure_tunneling_connection_config
+                Crt::Allocator *allocator,
+                Aws::Crt::Io::ClientBootstrap *clientBootstrap,
+                Aws::Crt::Io::SocketOptions *socketOptions,
+
+                std::string accessToken,
+                aws_secure_tunneling_local_proxy_mode localProxyMode,
+                std::string endpointHost,
+
+                OnSecureTunnelingConnectionComplete onConnectionComplete,
+                OnSecureTunnelingSendDataComplete onSendDataComplete,
+                OnSecureTunnelingDataReceive onDataReceive,
+                OnSecureTunnelingStreamStart onStreamStart,
+                OnSecureTunnelingStreamReset onStreamReset,
+                OnSecureTunnelingSessionReset onSessionReset);
+            SecureTunnel(const SecureTunnel &) = delete;
+            SecureTunnel(SecureTunnel &&) noexcept;
+
+            virtual ~SecureTunnel();
+
+            SecureTunnel &operator=(const SecureTunnel &) = delete;
+            SecureTunnel &operator=(SecureTunnel &&) noexcept;
+
+            int Connect();
+
+            int Close();
+
+            int SendData(const Crt::ByteCursor &data);
+
+            int SendStreamStart();
+
+            int SendStreamReset();
+
+            aws_secure_tunnel *GetUnderlyingHandle();
+
+          private:
+            void free();
```

Comment:
This doesn't need to be a separate function, you can just call `this->~SecureTunnel()` in the move ctor.

Reply:
Done. I didn't know I can call the destructor directly. Thanks!